### PR TITLE
Update BuildsService to use toast notifications

### PR DESCRIPTION
**Pull request by spadgett**
Use consistent toast notification messages regardless of the page the build was started from.

https://trello.com/c/6WVJdgjL

### CODE REVIEW

File: app/scripts/services/builds.js [hunk 1]
```diff
@@ -1,39 +1,143 @@
...
+        NotificationsService.addNotification({
+          type: "success",
+          message: _.capitalize(buildType) + " " + originalDisplayName + " is being rebuilt as " + clonedDisplayName + ".",
+          links: getNewBuildLinks(clonedBuild)
+        });
+      }, function(result) {
+        NotificationsService.addNotification({
+          type: "error",
+          message: "An error occurred while rerunning " + buildType + " " + originalDisplayName + ".",
+          details: $filter('getErrorDetails')(result)
```
Comment (jwforres) [inline app/scripts/services/builds.js]: getErrorDetails instead of calling $filter

File: app/views/browse/build-config.html [hunk 2]
```diff
@@ -121,7 +121,7 @@ <h1 class="contains-actions">
 
                         <button
                             class="btn btn-primary btn-lg"
-                            ng-click="startBuild(buildConfig.metadata.name)"
+                            ng-click="startBuild()"
```
Comment (jwforres) [inline app/views/browse/build-config.html]: Should you be passing buildConfig here?

**Comment by spadgett** [comment]
> Based on the method signature change to startBuild I'm surprised there were no changes required in the overview templates

Yeah, `overview.startBuild` took in a build config, so no change needed in the views.

File: app/views/browse/build-config.html (hunk 2 above)
Comment (spadgett) [inline app/views/browse/build-config.html]: The controller uses `$scope.buildConfig` when it calls `BuildsService.startBuild`, so this is OK.

**Comment by spadgett** [comment]
@jwforres thanks, updated

**Comment by jwforres** [comment]
[merge]

**Pull request merged**